actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -1174,45 +1360,63 @@ def permitted_scalar_filter(params, permitted_key)
         end
       end
 
-      def array_of_permitted_scalars?(value)
-        if value.is_a?(Array) && value.all? { |element| permitted_scalar?(element) }
-          yield value
-        end
-      end
-
       def non_scalar?(value)
         value.is_a?(Array) || value.is_a?(Parameters)
       end
 
       EMPTY_ARRAY = [] # :nodoc:
       EMPTY_HASH  = {} # :nodoc:
-      def hash_filter(params, filter)
+      def hash_filter(params, filter, on_unpermitted: false, explicit_arrays: false)
```

Comment:
Default values should match default values of `def permit_filters`, i.e. `def hash_filter(params, filter, on_unpermitted: nil, explicit_arrays: true)`

Reply:
I partly disagree. `hash_filter` existed before this, so its the method signature should be defined in a way that it keeps its original behavior. `permit_filters` is new and defines its defaults as the preferred usage in keeping with `expect`.

I partly agree though because I think this means the default should actually be `on_unpermitted: self.class.action_on_unpermitted_parameters`.

Reply:
You have a point here👍